databuilder/transformer/elasticsearch_document_transformer.py

```diff
@@ -12,13 +13,18 @@ class ElasticsearchDocumentTransformer(Transformer):
     """
     ELASTICSEARCH_INDEX_CONFIG_KEY = 'index'
     ELASTICSEARCH_DOC_CONFIG_KEY = 'doc_type'
+    ELASTICSEARCH_RESOURCE_CONFIG_KEY = 'resource_type'
 
     def init(self, conf):
         # type: (ConfigTree) -> None
         self.conf = conf
 
         self.elasticsearch_index = self.conf.get_string(ElasticsearchDocumentTransformer.ELASTICSEARCH_INDEX_CONFIG_KEY)
         self.elasticsearch_type = self.conf.get_string(ElasticsearchDocumentTransformer.ELASTICSEARCH_DOC_CONFIG_KEY)
+        # values: table, user
```

Comment:
I think resource type is something expected to grow. How about having enum on this?

Reply:
Tao,
Could we have this as enum? We can improve other part later on, but I think this one is expected to grow and it's good to have enum from the beginning.

Reply:
I am not sure if it is easy to do given the resource type is  pass from airflow databuilder dag as a config. If we make an enum, we need to put the enum into a library to be read by the databuilder dag as well. Currently I made a set(https://github.com/lyft/amundsendatabuilder/blob/70fdb1f42b692dca77cf13efcda5bbea53f4adde/databuilder/transformer/elasticsearch_document_transformer.py#L17) which defined all the available resource and throw exception if the resource is not part of the set. (edited) 
let me know what you think :)